File: Day2/Day2.java
import java.util.*;

public class Day2 {
    public void run() {
        Scanner sc = new Scanner(System.in);
        ArrayList<String> input = new ArrayList<>();
        while (sc.hasNext()) {
            input.add(sc.next());
        }

        int hasTwo = count(input, 2);
        int hasThree = count(input, 3);

        System.out.println("Result: " + hasTwo * hasThree);
    }

    private int count(ArrayList<String> in, int total) {
        int totalRes = 0;
        for (String s : in) {
            int res = 0;
            HashMap<Character, Integer> count = new HashMap<>();
            for (int i = 0; i < s.length(); i++) {
                char c = s.charAt(i);
                if (!count.containsKey(c)) {
                    count.put(c, 1);
                    continue;
                }

                if (count.get(c) == total-1) {
                    res++;
                } else if (count.get(c) == total) {
                    res--;
                } 
                count.put(c, count.get(c) + 1);

            }
            
            if (res > 0) totalRes++;
        }
        return totalRes;
    }

    public static void main(String[] args) {
        Day2 d = new Day2();
        d.run();
    }
}

File: Day2/Day2.cpp
#include <iostream>
#include <cstdio>
#include <unordered_map>
#include <vector>
#include <string>

using namespace std;

/** Count the total number of strings in in that have at least one character with exactly
    total occurrences. */
int count(vector<string> in, int total)
{
    int totalRes = 0;
    for (vector<string>::iterator it = in.begin(); it != in.end(); ++it)
    {
        string s = *it;
        int res = 0; // The number of characters that have exactly total occurrences
        unordered_map<char, int> count; // Number of occurences of a char

        for (int i = 0; i < s.size(); ++i)
        {
            // If count does not contain char at s[i]
            if (count.find(s[i]) == count.end())
            {
                count[s[i]] = 1; 
                continue;
            }

            // Only add to res when it hits the total required
            if (count[s[i]] == total - 1) 
            {
                ++res;
            } 
            // Decrement res if it exceeds total required
            else if (count[s[i]] == total) 
            {
                --res;
            }
            // Add to the count
            ++count[s[i]];
        }

        // Add to the total result
        if (res > 0) ++totalRes; 
    }
    return totalRes;
}

int main()
{
    // Input routine
    vector<string> input;
    string s;
    while (getline(cin, s))
    {
        if (s.empty()) break;
        input.push_back(s);
    }

    // Generate result
    int hasTwo = count(input, 2);
    int hasThree = count(input, 3);
    cout << (hasTwo * hasThree) << endl;
}